### DISCUSSION TITLE
exit code isn't picked up for me - am I testing this wrong?

### DISCUSSION THREAD

**Discussion by Lockszmith-GH** [JanDeDobbeleer/oh-my-posh#2842] [Troubleshoot] [answered]
This seems like a bug to me, but I don't see people reporting it, so doubting myself here a bit.

Problem: exit code is not picked up.

How I'm testing:
```
POSH_CONFIG=<poshconfigfile>
# Expecting 'exit(false)'
bash -c "exit 1" ; oh-my-posh --config $POSH_CONFIG debug | grep "^exit("
# Expecting 'exit(true)'
bash -c "exit 0" ; oh-my-posh --config $POSH_CONFIG debug | grep "^exit("
```

Depending on the configuration I either get all `false` or all `true` - usually all `true`.

Where I'm running:
Testing with `bash`, on Debian bullseye or Ubuntu Jammy based systems (Pop!_OS, TrueNAS SCALE)

My question:
How is the exit code collected, am I missing something? Am I not testing correctly?

**Comment by JanDeDobbeleer** [comment, marked as answer]
@Lockszmith-GH the test isn't going to work as the [init script](https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/src/shell/scripts/omp.bash) is where that magic happens. So without initializing in the shell using `eval "$(oh-my-posh init bash)"` this will never work correctly.

**Reply by Lockszmith-GH** [reply]
Thanks for setting me straight. I had some history manipulation running as part of the PROMPT hook before `_omp_hook`.
Your comment helped me identify it.
